src/Control/Carrier/ContainerRegistryApi.hs

```diff
@@ -251,31 +253,41 @@ exportBlob ::
   (RepoDigest, Bool, Text) ->
   m (Path Abs File)
 exportBlob manager imgSrc dir (digest, isGzip, targetFilename) = do
-  ctx <- ask
-  let sinkTarget :: Path Abs File
-      sinkTarget = dir </> Path (toString targetFilename)
-
-  let imgSrc' = imgSrc{registryContainerRepositoryReference = RepoReferenceDigest digest}
-
-  -- Prepare request with necessary authorization
-  req <- blobEndpoint imgSrc'
-  token <- getAuthToken (registryCred imgSrc) req manager Nothing =<< getToken ctx
-  let req' = applyAuthToken token req
-
-  -- Download image artifact
-  sendIO . runResourceT $ do
-    response <- HTTPConduit.http req' manager
-    runConduit $
-      HTTPConduit.responseBody response
-        .| (if isGzip then ungzip else idC)
-        .| sinkFile (toString sinkTarget)
-
-  logInfo . pretty $
-    if isGzip
-      then "Gzip extracted & downloaded: " <> targetFilename
-      else "Downloaded: " <> targetFilename
-
-  pure sinkTarget
+  exportJobId <- sendIO UUID.nextRandom
```

Comment:
I've changed very little in this snippet. Mainly indentation changed because of the new `context` call.